### PR TITLE
Add IM integration with create and counters

### PR DESCRIPTION
**Pull request by christianmoretti**
Hi @abrom can you take a look? Thanks.

### CODE REVIEW

File: docs/im.md [hunk 1]
```diff
@@ -0,0 +1,27 @@
+### IM API
+
+RocketChat's Im Messages.
```
Comment (abrom) [inline docs/im.md]: Should probably be `RocketChat's IM interface.`  (IM is abbreviation for instance message)

File: docs/im.md [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+#### im.create
+
+Create a direct message.
+
+```ruby
+require 'rocketchat'
+
+rocket_server = RocketChat::Server.new('http://your.server.address/')
+session = rocket_server.login('username', 'password')
+im = session.im.create(username: 'rocket.cat')
```
Comment (abrom) [inline docs/im.md]: it looks like the `create` endpoint returns a `room` and not an `im`

File: docs/im.md [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+#### im.counters
+
+Retrieves the count information using room_id *or* username
+
+```ruby
+require 'rocketchat'
+
+rocket_server = RocketChat::Server.new('http://your.server.address/')
+session = rocket_server.login('username', 'password')
+im = session.im.counters(room_id: 'room_id', username: 'rocket.cat')
```
Comment (abrom) [inline docs/im.md]: i suspect this should be a hash of the result (it's also not an `im`)

File: lib/rocket_chat/messages/im.rb [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+      # @param [String] room_id Rocket.Chat roomId
+      # @param [String] username Rocket.Chat username
+      # @return [RocketChat::Im]
+      # @raise [HTTPError, StatusError]
+      #
+      def counters(room_id: nil, username: nil)
+        response = session.request_json(
+          "/api/v1/im.counters?roomId=#{room_id}&username=#{username}"
+        )
+        RocketChat::Im.new response
```
Comment (abrom) [inline lib/rocket_chat/messages/im.rb]: Hmm not so sure about this. The `Im` model name would suggest that it is actually a message itself being returned. Maybe `ImSummary` would be more appropriate?  (or just have it a hash instead?)

File: lib/rocket_chat/messages/im.rb [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+        @session = session
+      end
+
+      #
+      # im.create REST API
+      # @param [String] username Rocket.Chat username
+      # @return [RocketChat::Room]
+      # @raise [HTTPError, StatusError]
+      #
+      def create(username: nil)
```
Comment (abrom) [inline lib/rocket_chat/messages/im.rb]: given `username` is required, why specify a default? Should be `create(username:)`

File: lib/rocket_chat/messages/im.rb [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+      end
+
+      #
+      # im.counters REST API
+      # @param [String] room_id Rocket.Chat roomId
+      # @param [String] username Rocket.Chat username
+      # @return [RocketChat::Im]
+      # @raise [HTTPError, StatusError]
+      #
+      def counters(room_id: nil, username: nil)
```
Comment (abrom) [inline lib/rocket_chat/messages/im.rb]: given `roomId` is required, this should be `counters(room_id:, username: nil)`

File: lib/rocket_chat/messages/im.rb [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+        response = session.request_json(
+          "/api/v1/im.counters?roomId=#{room_id}&username=#{username}"
```
Comment (abrom) [inline lib/rocket_chat/messages/im.rb]: would prefer you use the existing mechanism for specifying query params:
```
response = session.request_json(
  '/api/v1/im.counters',
  body: {
    roomId: room_id,
    username: username
  }
)```

File: spec/rocket_chat/messages/im_spec.rb [hunk 8]
```diff
@@ -0,0 +1,149 @@
...
+            success: false,
+            error: 'Body param "roomId" or "username" is required [error-room-param-not-provided]',
+            errorType: 'error-room-param-not-provided'
+          }.to_json,
+          status: 400
+        )
+    end
+
+    context 'with a valid session' do
+      it 'get qty of messages' do
```
Comment (abrom) [inline spec/rocket_chat/messages/im_spec.rb]: Please don't use abbreviations in spec descriptions. They should be as readable as possible!

File: lib/rocket_chat/im.rb [hunk 9]
```diff
@@ -0,0 +1,54 @@
+module RocketChat
+  #
+  # Rocket.Chat Info
+  #
+  class Im
```
Comment (abrom) [inline lib/rocket_chat/im.rb]: Seems to be some coverage missing for this class

File: spec/rocket_chat/messages/im_spec.rb [hunk 10]
```diff
@@ -0,0 +1,149 @@
...
+        end.to raise_error RocketChat::StatusError
+      end
+    end
+  end
+
+  describe '#counters' do
+    before do
+      stub_unauthed_request :get, '/api/v1/im.counters'
+
+      stub_authed_request(:get, '/api/v1/im.counters?roomId=rocket.cat&username=')
```
Comment (abrom) [inline spec/rocket_chat/messages/im_spec.rb]: you don't appear to be testing any case where the `username` property is provided?

File: docs/im.md (hunk 1 above)
Comment (christianmoretti) [reply docs/im.md]: Ok, updated.

File: docs/im.md (hunk 2 above)
Comment (christianmoretti) [reply docs/im.md]: Update too.

File: lib/rocket_chat/messages/im.rb [hunk 11]
```diff
@@ -0,0 +1,52 @@
...
+          body: { username: username }
+        )
+        RocketChat::Room.new response['room']
+      end
+
+      #
+      # im.counters REST API
+      # @param [String] room_id Rocket.Chat roomId
+      # @param [String] username Rocket.Chat username
+      # @return [RocketChat::Im]
```
Comment (abrom) [inline lib/rocket_chat/messages/im.rb]: should be ... `ImSummary`

File: lib/rocket_chat/im_summary.rb [hunk 12]
```diff
@@ -0,0 +1,54 @@
+module RocketChat
+  #
+  # Rocket.Chat Info
```
Comment (abrom) [inline lib/rocket_chat/im_summary.rb]: `Rocket.Chat IM Summary` is probably more appropriate

**Comment by abrom** [comment]
Released in v0.1.18

**Pull request merged**